### PR TITLE
 koreader: 2022.08 -> 2023.04 

### PR DESCRIPTION
**Pull request by symphorien**
###### Description of changes

[koreader: fix startup](https://github.com/NixOS/nixpkgs/commit/dfd449b9588fae491e5d92d565e1e2e2472d6c5e)

koreader looks for fonts/NotoSans-Regular.ttf at startup but our version
of noto fonts provide different files under different names (eg
NotoSans[wdth,wght].ttf) so koreader crashes.

Let's use the font files provided in the source repo instead. (they are
not in the debian package it seems)

[koreader: 2022.08 -> 2023.04](https://github.com/NixOS/nixpkgs/commit/e12f96e61edad2f8011a2b92c241d19e191f2e58)

changelogs:
https://github.com/koreader/koreader/releases/tag/v2022.10
https://github.com/koreader/koreader/releases/tag/v2022.11
https://github.com/koreader/koreader/releases/tag/v2023.01
https://github.com/koreader/koreader/releases/tag/v2023.03
https://github.com/koreader/koreader/releases/tag/v2023.04

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by wegank** [comment]
#229766

File: pkgs/applications/misc/koreader/default.nix [hunk 1]
```diff
@@ -37,16 +46,19 @@ in stdenv.mkDerivation rec {
   dontBuild = true;
 
   installPhase = ''
+    # like ln -s but checks that the pointee exists
```
Comment (wegank) [inline pkgs/applications/misc/koreader/default.nix:49]: Which one doesn't?

**Review by contrun — APPROVED** [review]
Tried `nix run github:symphorien/nixpkgs/e12f96e#koreader` . It works. Thanks.

File: pkgs/applications/misc/koreader/default.nix (hunk 1 above)
Comment (symphorien) [reply pkgs/applications/misc/koreader/default.nix:49]: In the current expression `ln -s "${font-droid}/share/fonts/opentype/NerdFonts/Droid Sans Mono Nerd Font Complete Mono.otf"` makes a broken symlink, and most other font related symlink. In the end I kept none of them, so maybe it's overzealous.

File: pkgs/applications/misc/koreader/default.nix [hunk 2]
```diff
@@ -1,24 +1,33 @@
...
+  luajit_lua52 = luajit.override { enable52Compat = true; };
+in
+stdenv.mkDerivation rec {
   pname = "koreader";
-  version = "2022.08";
+  version = "2023.04";
 
   src = fetchurl {
     url =
       "https://github.com/koreader/koreader/releases/download/v${version}/koreader-${version}-amd64.deb";
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/koreader/default.nix:21]:
```suggestion
    url = "https://github.com/koreader/koreader/releases/download/v${version}/koreader-${version}-amd64.deb";
```

File: pkgs/applications/misc/koreader/default.nix [hunk 3]
```diff
@@ -1,24 +1,33 @@
...
-    sha256 = "sha256-+JBJNJTAnC5gpuo8cehfe/3YwGIW5iFA8bZ8nfz9qsk=";
+    sha256 = "sha256-tRUeRB1+UcWT49dchN0YDvd0L5n1YRdtMSFc8yy6m5o=";
+  };
+
+  src_repo = fetchFromGitHub {
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/koreader/default.nix:25]: We can't build from source?

File: pkgs/applications/misc/koreader/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/applications/misc/koreader/default.nix:49]: We can't fix that with some clever find filter?

File: pkgs/applications/misc/koreader/default.nix (hunk 3 above)
Comment (symphorien) [reply pkgs/applications/misc/koreader/default.nix:25]: their build system is a shell script that calls a Makefile that calls cmake that fetches more than a dozen git repos during build: if it's possible, it's very hard.

File: pkgs/applications/misc/koreader/default.nix (hunk 1 above)
Comment (symphorien) [reply pkgs/applications/misc/koreader/default.nix:49]: If you have a better solution I'm all ears.

Comment (SuperSandro2000) [reply pkgs/applications/misc/koreader/default.nix:49]: After the update mkLink is used on three static files. Either they exist or they don't. I don't think we need a wrapper for this.

Comment (symphorien) [reply pkgs/applications/misc/koreader/default.nix:49]: I removed the wrapper.

**Comment by uninsane** [comment]
pardon the bump: i saw discussion about a from-source build in this PR so if anyone wants to pursue that i've got a working version over [here](https://github.com/nix-community/nur-combined/blob/0d2780d416a8d4109e906450787724d9305f27c7/repos/colinsane/pkgs/additional/koreader-from-src/default.nix). its approach is to include all of KOreader's dependencies in `srcs` and then let KOreader build everything from source as it ordinarily would, just without net requests.

i suspect it needs an update script for that approach to be maintainable, and i haven't inspected the closure to make sure it's not holding onto any refs that it shouldn't. not sure i'll get around to those things anytime soon, but if anyone else has the drive to take it across the finish line i'm happy to assist/review/test.

**Pull request merged**